2016-2017_Starstruck/Module Test Code/FileIO_Test/src/AutonRecorder.c

```diff
@@ -0,0 +1,193 @@
+/*
+ *  AutonRecorder.c
+ *
+ *  Created on: Feb 8, 2017
+ *      Author: Kirsten Meidlinger
+ */
+
+ #include <main.h>
+
+/*
+ * This file abstracts all of the functionality of 750E's Auton Recorder.
+ *   This file should be placed in the src directory. Note that it won't
+ *   compile without AutonRecorder.h properly included in main.h.
+ */
+
+  int slot;
+  TaskHandle recordTask;
+  Frame root;
+  char *fileName;
+
+  void startRecording(int s){
+    slot = s;
+    recordTask = taskCreate(recordingLoop,TASK_DEFAULT_STACK_SIZE,NULL,TASK_PRIORITY_DEFAULT);
+    root = NULLFRAME;
+    fileName = "";
+  }
+
+  void stopRecording(){
+    taskDelete(recordTask);
+    Frame currentFrame = root;
+    sprintf(fileName,"Recording%d.txt",slot);
+    FILE* recording = fopen(fileName,"w");
+    while(currentFrame.next != NULL){
+      fprintf(recording,frameToString(currentFrame));
+    }
+    fclose(recording);
+  	printf("File %s written and closed.\n",fileName);
+  }
+
+  Frame loadRecording(int s){
+    FILE* f = fopen(fileName,"r");
+    char *frame = malloc(50);
+    Frame rt = NULLFRAME;
+    while(fgets(frame,50,f) != NULL){
+      if(rt.analog_main[0] == -1){
+        //rt =
+      }
+    }
+    /*
+    //TEST CODE FOR READING A FILE
+
+    //Initialize 20 char string for read
+    char *a = malloc(sizeof(char) * 10);
+    char *b = malloc(sizeof(char) * 10);
+    char *c = malloc(sizeof(char) * 10);
+    //Open Test file in read mode
+    f = fopen("Test.txt","r");
+    //Read file contents to str
+    fgets(a, 4, f); //string, max chars + 1, file
+    fgets(b, 4, f);
+    fgets(c, 4, f);
+    //Close file
+    fclose(f);
+    printf("%d, %d, %d\n\r",atoi(a),atoi(b),atoi(c));
+    */
+
+    return root; //TODO: placeholder
+  }
+
+  void recordingLoop(){
+    if(root.analog_main[0] == -1){ //checks if it's equal to NULLFRAME
```

Comment:
any reason you did this here and not in startRecording? Might actually be better this way but I was wondering if you had a specific reason